dockstore-webservice/src/main/java/io/dockstore/webservice/resources/DockerRepoTagResource.java

```diff
@@ -246,4 +259,30 @@ private Tool findToolByIdAndCheckToolAndUser(Long toolId, User user) {
         checkUser(user, tool);
         return tool;
     }
+
+    @GET
+    @Timed
+    @UnitOfWork(readOnly = true)
+    @Path("{containerId}/tags/{tagId}/sourcefiles")
+    @Operation(operationId = "getTagsSourcefiles", description = "Retrieve sourcefiles for a container's version", security = @SecurityRequirement(name = OPENAPI_JWT_SECURITY_DEFINITION_NAME))
+    public SortedSet<SourceFile> getTagsSourceFiles(@ApiParam(hidden = true) @Auth Optional<User> user,
+            @Parameter(name = "containerId", description = "Container to retrieve the version from", required = true, in = ParameterIn.PATH) @PathParam("containerId") Long containerId,
+            @Parameter(name = "tagId", description = "Tag to retrieve the sourcefiles from", required = true, in = ParameterIn.PATH) @PathParam("tagId") Long tagId,
+            @Parameter(name = "fileTypes", description = "List of file types to filter sourcefiles by") @QueryParam("fileType") List<DescriptorLanguage.FileType> fileTypes) {
+        Tool tool = toolDAO.findById(containerId);
+        checkEntry(tool);
+        checkOptionalAuthRead(user, tool);
+
+        Version version = tagDAO.findVersionInEntry(containerId, tagId);
+        if (version == null) {
+            throw new CustomWebApplicationException("Container tag " + tagId + " does not exist for this tool", HttpStatus.SC_BAD_REQUEST);
+        }
+
+        SortedSet<SourceFile> sourceFiles = version.getSourceFiles();
```

Comment:
I only thought of this after you responded to my last comment, but you _could_ just query the source files from the database directly. Any time we are filtering Hibernate objects, it's an indicator of a potential area where you could have the DB do the work for you.

Is the common use case in the UI expected to be invoking going to be using the filetypes parameter, and if so, what type? Will it just be the .dockstore.yml? If that is a common case, it's a slightly stronger argument to fetch from the DB; if a workflow has n descriptor files and one .dockstore.yml, just fetch the .dockstore.yml from the DB.

This is probably way overkill, and we risk having too many one-offs, so I'm leaning towards not doing it, but throwing it out for consideration.